### PR TITLE
fix final problems caused by the rebase to c8ac18d

### PR DESCRIPTION
**Pull request by marcrasi**
These fixes make everything compile, and all the test pass.

I'll add github comments in the PR explaining the reasons for some of the changes.

(Note that this PR is into a `tensorflow-rebase` branch, not into `tensorflow`, so I'll need to do one final step of moving `tensorflow-rebase` into `tensorflow` after I land this PR).

### CODE REVIEW

File: lib/SILOptimizer/Mandatory/MandatoryInlining.cpp
Comment (marcrasi) [inline lib/SILOptimizer/Mandatory/MandatoryInlining.cpp]: I moved these checks into the `shouldInlinePredicate` function, so that we can have different checks for TF and for non-TF inlining.

This is probably not the most correct solution, but it makes all the tests pass.

File: utils/update-checkout-config.json
Comment (dan-zheng) [inline utils/update-checkout-config.json]: Minor: would you mind reordering these to be in the original order?

File: utils/update-checkout-config.json
Comment (marcrasi) [inline utils/update-checkout-config.json]: I pointed these at my forks of the repos, because I don't have commit access to the original repos yet. I'll point these back to the original repos before I merge this PR.

File: test/TensorFlow/integration.swift
Comment (marcrasi) [inline test/TensorFlow/integration.swift]: I am guessing that these differences are caused by the switch to guaranteed calling, but could someone confirm that this is correct?

File: test/TensorFlow/retain_release.swift
Comment (marcrasi) [inline test/TensorFlow/retain_release.swift]: Same.

File: lib/AST/ASTPrinter.cpp
Comment (marcrasi) [inline lib/AST/ASTPrinter.cpp]: I cherry-picked this change from master, because it makes a failing test pass.

File: lib/SIL/SILPrinter.cpp
Comment (marcrasi) [inline lib/SIL/SILPrinter.cpp]: This is just part of the cherry-picked change that I mention above.

File: utils/update-checkout-config.json
Comment (marcrasi) [reply utils/update-checkout-config.json]: done

File: lib/SILOptimizer/Mandatory/TFPartition.cpp
Comment (rxwei) [inline lib/SILOptimizer/Mandatory/TFPartition.cpp]: I think "noNestedConflict" is what's in the function signature.

File: lib/SILOptimizer/Mandatory/TFPartition.cpp
Comment (rxwei) [inline lib/SILOptimizer/Mandatory/TFPartition.cpp]: "noNestedConflict"

File: lib/SILOptimizer/Mandatory/TFPartition.cpp
Comment (rxwei) [inline lib/SILOptimizer/Mandatory/TFPartition.cpp]: "noNestedConflict"

File: lib/SILOptimizer/Mandatory/TFPartition.cpp
Comment (rxwei) [inline lib/SILOptimizer/Mandatory/TFPartition.cpp]: "noNestedConflict"

**Review by rxwei — APPROVED** [review]

File: lib/SILOptimizer/Mandatory/MandatoryInlining.cpp
Comment (slavapestov) [reply lib/SILOptimizer/Mandatory/MandatoryInlining.cpp]: If TF mandatory inlining needs to inline non-transaprent functions, please update the SIL linker in lib/SIL/Linker.cpp and lib/SILOptimizer/UtilityPasses/Link.cpp.

File: test/TensorFlow/integration.swift
Comment (mhong) [reply test/TensorFlow/integration.swift]: Can you provide the full SIL code?

File: test/TensorFlow/retain_release.swift
Comment (mhong) [inline test/TensorFlow/retain_release.swift]: a heads-up that i've seen piper behave differently in this regard. one option is to replace with {{.*}}

File: test/TensorFlow/retain_release.swift
Comment (mhong) [reply test/TensorFlow/retain_release.swift]: Yes this should be fine as it's add and removing the same # retain/release insts.

**Review by mhong — COMMENTED** [review]
Changes in TFUtilities.cpp and TFPartition.cpp look good.

There's one unit test change that I commented out, and can look at more when you provide the full SIL code.

File: test/TensorFlow/integration.swift
Comment (marcrasi) [reply test/TensorFlow/integration.swift]: here's the SIL for the TFPartition Host Result for this function: https://gist.github.com/marcrasi/b2fdaeb542b9b9d307bb49efe14e1406

I also included the TFPartition Host Result for the function that I changed in retain_release.swift, in case you want to look at that.

File: test/TensorFlow/retain_release.swift
Comment (marcrasi) [reply test/TensorFlow/retain_release.swift]: FileCheck doesn't require that a `CHECK` matches the entire line, so it should already work no matter what stuff appears in front.

File: lib/SILOptimizer/Mandatory/TFPartition.cpp
Comment (marcrasi) [reply lib/SILOptimizer/Mandatory/TFPartition.cpp]: done (all of them)

**Review by mhong — APPROVED** [review]

**Review by lattner — APPROVED** [review]
LGTM with minor formatting changes, thank you!

File: lib/SILOptimizer/Mandatory/MandatoryInlining.cpp
Comment (marcrasi) [reply lib/SILOptimizer/Mandatory/MandatoryInlining.cpp]: Done: https://bugs.swift.org/browse/SR-8015

I'll also add a todo in the code here referencing the issue.

**Pull request merged**
